src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,41 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
+import { UserDTO } from './user.dto';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
   constructor(private userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param('id') id) {
+    const errResp: IResult<UserDTO> = {
+      status: 'NOT OK',
+      error: '',
+      result: null,
+    };
+
+    try {
+      const user = await this.userService.getUserById(id);
+      if (user) {
+        const userResp: UserDTO = {
+          name: user.name,
+          _id: user.id,
```

Comment:
Did you check if this is working? Mongo's objects have `_id` instead of `id` I guess?

Reply:
yeah we need to convert to ObjectId i guess

Reply:
Using id instead of _id works in case of MongoDB. Its like an alias afaik. But it sometimes may throw an illegal error. Too used to using id. Changed it to _id anyway now.